### PR TITLE
Fix nav bar for not signed in user and tests

### LINKED ISSUE #437: change `@box_requests.where...` to  `@box_requests = @box_requests.where...` ?
@sarslanoglu should this be `@box_requests = @box_requests.where...` ?

_Originally posted by @maebeale in https://github.com/rubyforgood/voices-of-consent/pull/414_

### PR DESCRIPTION
**Pull request by sarslanoglu**
Fix not signed_in user nav bar and its tests
Also closes #437

### CODE REVIEW

File: app/views/shared/_navigation.html.erb [hunk 1]
```diff
@@ -51,6 +51,27 @@
         </a>
       <% end %>
     <% end %>
+  <% else %>
```
Comment (maebeale) [inline app/views/shared/_navigation.html.erb:54]: @sarslanoglu I thought we'd decided unauthenticated visitors to the site would only see the home/logo button and the login?

File: app/controllers/box_requests_controller.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def index
     if params[:filter_by].present?
       (filter_attr = params[:filter_by])
       @box_requests = @box_requests.public_send(filter_attr)
-      @box_requests.where(reviewed_by_id: params[:current_user_id]) if params[:current_user_id].present?
+      @box_requests = @box_requests.where(reviewed_by_id: params[:current_user_id]) if params[:current_user_id].present?
```
Comment (maebeale) [inline app/controllers/box_requests_controller.rb:12]: @sarslanoglu - thank you!

Comment (h-m-m) [inline app/controllers/box_requests_controller.rb]: Is it safe to make this stricter? For example, if there is supposed to be no way to get here without logging in, I'd be happy if this raises an exception when `params[:current_user_id]` is missing

Comment (h-m-m) [inline app/controllers/box_requests_controller.rb:12]: Nice job catching this mistake. Do you mind adding a test to make sure nobody else accidentally changes the code to revert to the previous behavior?

Also, is it safe to make this stricter? For example, if there is supposed to be no way to get here without logging in, I'd be happy if this raises an exception when `params[:current_user_id]` is missing.

File: app/views/shared/_navigation.html.erb [hunk 3]
```diff
@@ -51,6 +51,27 @@
...
+    <a class="nav-icon">
+      <img src="https://voices-of-consent-static-images.s3.amazonaws.com/Status.svg" title="Status">
```
Comment (h-m-m) [inline app/views/shared/_navigation.html.erb:56]: Having the full nav bar not appear if nobody is logged in might be a good thing. I kinda liked how your previous PR did that. Any opinions on this, @maebeale?

**Review by h-m-m — COMMENTED** [review]
Wow, thanks @sarslanoglu for responding so quickly! I appreciate your help with this project.

I had a few questions about your code, when you have a moment

**Comment by maebeale** [comment]
@h-m-m I vote we remove the links unless authenticated

**Review by maebeale — APPROVED** [review]

**Pull request merged**